### ISSUE TITLE
memory usage of symbolizer's linerCache?

### ISSUE THREAD

**Issue by dreamerlzl** [parca-dev/parca#4309]
Judging from my understanding of the code and the memory profile of parca, it seems that there are no configurations/controls for the memory usage of symbolization caching:

https://github.com/parca-dev/parca/blob/6dae29a493dfab35448ea2324c07c60aafbf50ad/pkg/symbol/elfutils/debuginfofile.go#L43

Currently I don't know how to set a memory limit for parca properly so thanks for any suggestion/insight.

**Comment by dreamerlzl** [comment]
![Uploading Screenshot 2024-02-19 at 14.01.47.png…]()

**Comment by brancz** [comment]
I think we just need to change the whole symbolization strategy here. We can do all of this entirely without the `godwarf.LoadTree` stuff and it'll make everything way faster, at that point we probably won't even need the liner cache anymore.

**Comment by dreamerlzl** [comment]
> I think we just need to change the whole symbolization strategy here. We can do all of this entirely without the `godwarf.LoadTree` stuff and it'll make everything way faster, at that point we probably won't even need the liner cache anymore.

Thanks for the update.  So we can do nothing here to limit the memory growing, right?

**Comment by brancz** [comment]
That's right, but I hope to get to working on this soon, as right now it's quite terrible as it depends on the binaries involved how much memory is used, that shouldn't be the case, and it's definitely possible to change this!
